src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java

```diff
@@ -2089,22 +2096,17 @@ private void paintCells(Graphics g, int rMin, int rMax, int cMin, int cMax) {
                 }
             }
         } else {
-            for(int row = rMin; row <= rMax; row++) {
+            for (int row = rMin; row <= rMax; row++) {
                 cellRect = table.getCellRect(row, cMin, false);
-                aColumn = cm.getColumn(cMin);
-                if (aColumn != draggedColumn) {
-                    columnWidth = aColumn.getWidth();
-                    cellRect.width = columnWidth - columnMargin;
-                    paintCell(g, cellRect, row, cMin);
-                }
-                for(int column = cMin+1; column <= cMax; column++) {
+                cellRect.x = getXPosition(cMax);
+                for (int column = cMax; column >= cMin; column--) {
```

Comment:
Is there any reason for choosing to paint from cMax to cMin ie from left-to-right in RTL orientation
when it was done from right-to-left before your fix? 
I guess painting from right-to-left also should work with cellRect.x -= columnWidth ie with no change in code, no?

One more q,
If there are 5 columns, then is it that for LTR cMin is 1, cMax 5 and for RTL cMin is 5 , cMax 1 or viceversa?

Reply:
Yes, we can either paint from right-to-left or from left-to-right. I followed `JTableHeader ` paint logic.  for 5 columns cMin = 0 and cMax = 4.

Reply:
For both LTR and RTL?

Reply:
Yes.